include/aikido/planner/ConfigurationToConfigurations.hpp

```diff
@@ -0,0 +1,73 @@
+#ifndef AIKIDO_PLANNER_PLANTOCONFIGURATIONS_HPP_
+#define AIKIDO_PLANNER_PLANTOCONFIGURATIONS_HPP_
+
+#include "aikido/constraint/Testable.hpp"
+#include "aikido/planner/Problem.hpp"
+#include "aikido/statespace/Interpolator.hpp"
+#include "aikido/statespace/StateSpace.hpp"
+
+namespace aikido {
+namespace planner {
+
+/// Planning problem to plan to a multiple goal configurations.
+///
+/// Plan a trajectory from start state to any of the goal states using an
+/// interpolator to interpolate between the states.
+class ConfigurationToConfigurations : public Problem
+{
+public:
+  /// Constructor.
+  ///
+  /// \param stateSpace State space.
+  /// \param startState Start state.
+  /// \param goalStates Goal states.
+  /// \param interpolator Interpolator used to produce the output trajectory.
+  /// \param constraint Trajectory-wide constraint that must be satisfied.
+  /// \throw If \c stateSpace is not compatible to \c constraint's state space.
+  ConfigurationToConfigurations(
+      statespace::StateSpacePtr stateSpace,
+      const statespace::StateSpace::State* startState,
+      const std::vector<statespace::StateSpace::State*>& goalStates,
```

Comment:
I agree with @brianhou that we would need a set for fast lookup. From a performance perspective, the whole point of having this feature, to my understanding, is to parallelly search for multiple goals and terminate when path to one of them is found. This implies, we would ideally not want to _iterate_ through each goal and plan. I think if done right, this would need a set and not a vector. Correct me if I am wrong.

Reply:
Agreed. Changed to `std::unordered_set` instead of `std::set` because the order doesn't matter here.